src/smallmatter/pathlib.py

```diff
@@ -94,6 +94,11 @@ def pathify(path: Union[str, Path, PathLike], *args, **kwargs) -> Path2:
 class S3Path(Path2):
     """A path to S3 object."""
 
+    def __init__(self, path):
+        super(S3Path, self).__init__()
+        self.fs = s3fs.S3FileSystem(anon=False)
+        self.s3path = path
+
```

Comment:
```suggestion
```

Reply:
`S3Path` should always rely on the `fs` object injected by `Path2`. Rationale: to allow caller to use custom credential, and to always propagate the custom `fs` object on child operations (e.g., `/`).